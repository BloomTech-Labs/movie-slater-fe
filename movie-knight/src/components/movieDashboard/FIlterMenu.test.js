import React from 'react';
import { render, fireEvent } from '@testing-library/react';
import FilterMenu from "./FilterMenu.js";

// test('FilterMenu renders correctly',() =>{
//     expect(render(<FilterMenu/>)).toMatchSnapshot();
//    });

test('FilterMenu displays', () => {
    const wrapper = render(<FilterMenu/>);
    wrapper.getByText(/sort by/i)
    wrapper.getByText(/most recent/i)
    wrapper.getByText(/oldest/i)
    wrapper.getByText(/a-z/i)
    wrapper.getByText(/z-a/i)
    wrapper.getByText('R')
    wrapper.getByText(/pg-13/i)
    wrapper.getByText('PG')
    wrapper.getByText('G')
    wrapper.getByText(/see results/i)
});
