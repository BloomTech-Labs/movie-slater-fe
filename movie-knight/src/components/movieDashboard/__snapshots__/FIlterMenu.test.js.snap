// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`FilterMenu correctly 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div>
        <div
          id="hamburger-menu"
        >
          <div
            class="linediv"
          >
            Filter
            <div
              class="linecon"
            >
              <div
                class="line1 black"
              />
              <div
                class="line1 white"
              />
              <div
                class="line2 black"
              />
              <div
                class="line2 white"
              />
              <div
                class="line3 black"
              />
              <div
                class="line3 white"
              />
            </div>
          </div>
        </div>
        <div
          class="menu-filter"
          id="filter"
        >
          <form
            class="filtering"
            id="ratingSelect"
          >
            <div
              class="movie-rating"
            >
              Rating
              <label>
                <input
                  checked=""
                  id="R"
                  name="stars"
                  type="checkbox"
                  value="R"
                />
                R
              </label>
              <label>
                <input
                  checked=""
                  name="stars"
                  type="checkbox"
                  value="PG-13"
                />
                PG-13
              </label>
              <label>
                <input
                  checked=""
                  name="stars"
                  type="checkbox"
                  value="PG"
                />
                PG
              </label>
              <label>
                <input
                  checked=""
                  name="stars"
                  type="checkbox"
                  value="G"
                />
                G
              </label>
            </div>
            <div
              class="movie-review"
            >
              <label>
                Review Rating
              </label>
              <div
                class="star-buttons"
              >
                <label>
                  <input
                    checked=""
                    name="stars"
                    type="checkbox"
                    value="1"
                  />
                  <span
                    class="icon-full"
                  >
                    ★
                  </span>
                  <span
                    class="icon"
                  >
                    ☆
                  </span>
                  <span
                    class="icon"
                  >
                    ☆
                  </span>
                  <span
                    class="icon"
                  >
                    ☆
                  </span>
                  <span
                    class="icon"
                  >
                    ☆
                  </span>
                </label>
                <label>
                  <input
                    checked=""
                    name="stars"
                    type="checkbox"
                    value="2"
                  />
                  <span
                    class="icon-full"
                  >
                    ★
                  </span>
                  <span
                    class="icon-full"
                  >
                    ★
                  </span>
                  <span
                    class="icon"
                  >
                    ☆
                  </span>
                  <span
                    class="icon"
                  >
                    ☆
                  </span>
                  <span
                    class="icon"
                  >
                    ☆
                  </span>
                </label>
                <label>
                  <input
                    checked=""
                    name="stars"
                    type="checkbox"
                    value="3"
                  />
                  <span
                    class="icon-full"
                  >
                    ★
                  </span>
                  <span
                    class="icon-full"
                  >
                    ★
                  </span>
                  <span
                    class="icon-full"
                  >
                    ★
                  </span>
                  <span
                    class="icon"
                  >
                    ☆
                  </span>
                  <span
                    class="icon"
                  >
                    ☆
                  </span>
                </label>
                <label>
                  <input
                    checked=""
                    name="stars"
                    type="checkbox"
                    value="4"
                  />
                  <span
                    class="icon-full"
                  >
                    ★
                  </span>
                  <span
                    class="icon-full"
                  >
                    ★
                  </span>
                  <span
                    class="icon-full"
                  >
                    ★
                  </span>
                  <span
                    class="icon-full"
                  >
                    ★
                  </span>
                  <span
                    class="icon"
                  >
                    ☆
                  </span>
                </label>
                <label>
                  <input
                    checked=""
                    name="stars"
                    type="checkbox"
                    value="5"
                  />
                  <span
                    class="icon-full"
                  >
                    ★
                  </span>
                  <span
                    class="icon-full"
                  >
                    ★
                  </span>
                  <span
                    class="icon-full"
                  >
                    ★
                  </span>
                  <span
                    class="icon-full"
                  >
                    ★
                  </span>
                  <span
                    class="icon-full"
                  >
                    ★
                  </span>
                </label>
              </div>
            </div>
            <div
              class="sorting"
            >
              <label>
                Sort By :
              </label>
              <label>
                <input
                  name="filter"
                  type="radio"
                  value="recent"
                />
                Most Recent
              </label>
              <label>
                <input
                  name="filter"
                  type="radio"
                  value="old"
                />
                Oldest
              </label>
              <label>
                <input
                  name="filter"
                  type="radio"
                  value="az"
                />
                A-Z
              </label>
              <label>
                <input
                  name="filter"
                  type="radio"
                  value="za"
                />
                Z-A
              </label>
            </div>
            <button
              class="results-btn"
            >
              See Results
            </button>
          </form>
        </div>
      </div>
    </div>
  </body>,
  "container": <div>
    <div>
      <div
        id="hamburger-menu"
      >
        <div
          class="linediv"
        >
          Filter
          <div
            class="linecon"
          >
            <div
              class="line1 black"
            />
            <div
              class="line1 white"
            />
            <div
              class="line2 black"
            />
            <div
              class="line2 white"
            />
            <div
              class="line3 black"
            />
            <div
              class="line3 white"
            />
          </div>
        </div>
      </div>
      <div
        class="menu-filter"
        id="filter"
      >
        <form
          class="filtering"
          id="ratingSelect"
        >
          <div
            class="movie-rating"
          >
            Rating
            <label>
              <input
                checked=""
                id="R"
                name="stars"
                type="checkbox"
                value="R"
              />
              R
            </label>
            <label>
              <input
                checked=""
                name="stars"
                type="checkbox"
                value="PG-13"
              />
              PG-13
            </label>
            <label>
              <input
                checked=""
                name="stars"
                type="checkbox"
                value="PG"
              />
              PG
            </label>
            <label>
              <input
                checked=""
                name="stars"
                type="checkbox"
                value="G"
              />
              G
            </label>
          </div>
          <div
            class="movie-review"
          >
            <label>
              Review Rating
            </label>
            <div
              class="star-buttons"
            >
              <label>
                <input
                  checked=""
                  name="stars"
                  type="checkbox"
                  value="1"
                />
                <span
                  class="icon-full"
                >
                  ★
                </span>
                <span
                  class="icon"
                >
                  ☆
                </span>
                <span
                  class="icon"
                >
                  ☆
                </span>
                <span
                  class="icon"
                >
                  ☆
                </span>
                <span
                  class="icon"
                >
                  ☆
                </span>
              </label>
              <label>
                <input
                  checked=""
                  name="stars"
                  type="checkbox"
                  value="2"
                />
                <span
                  class="icon-full"
                >
                  ★
                </span>
                <span
                  class="icon-full"
                >
                  ★
                </span>
                <span
                  class="icon"
                >
                  ☆
                </span>
                <span
                  class="icon"
                >
                  ☆
                </span>
                <span
                  class="icon"
                >
                  ☆
                </span>
              </label>
              <label>
                <input
                  checked=""
                  name="stars"
                  type="checkbox"
                  value="3"
                />
                <span
                  class="icon-full"
                >
                  ★
                </span>
                <span
                  class="icon-full"
                >
                  ★
                </span>
                <span
                  class="icon-full"
                >
                  ★
                </span>
                <span
                  class="icon"
                >
                  ☆
                </span>
                <span
                  class="icon"
                >
                  ☆
                </span>
              </label>
              <label>
                <input
                  checked=""
                  name="stars"
                  type="checkbox"
                  value="4"
                />
                <span
                  class="icon-full"
                >
                  ★
                </span>
                <span
                  class="icon-full"
                >
                  ★
                </span>
                <span
                  class="icon-full"
                >
                  ★
                </span>
                <span
                  class="icon-full"
                >
                  ★
                </span>
                <span
                  class="icon"
                >
                  ☆
                </span>
              </label>
              <label>
                <input
                  checked=""
                  name="stars"
                  type="checkbox"
                  value="5"
                />
                <span
                  class="icon-full"
                >
                  ★
                </span>
                <span
                  class="icon-full"
                >
                  ★
                </span>
                <span
                  class="icon-full"
                >
                  ★
                </span>
                <span
                  class="icon-full"
                >
                  ★
                </span>
                <span
                  class="icon-full"
                >
                  ★
                </span>
              </label>
            </div>
          </div>
          <div
            class="sorting"
          >
            <label>
              Sort By :
            </label>
            <label>
              <input
                name="filter"
                type="radio"
                value="recent"
              />
              Most Recent
            </label>
            <label>
              <input
                name="filter"
                type="radio"
                value="old"
              />
              Oldest
            </label>
            <label>
              <input
                name="filter"
                type="radio"
                value="az"
              />
              A-Z
            </label>
            <label>
              <input
                name="filter"
                type="radio"
                value="za"
              />
              Z-A
            </label>
          </div>
          <button
            class="results-btn"
          >
            See Results
          </button>
        </form>
      </div>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;

exports[`FilterMenu renders correctly 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div>
        <div
          id="hamburger-menu"
        >
          <div
            class="linediv"
          >
            Filter
            <div
              class="linecon"
            >
              <div
                class="line1 black"
              />
              <div
                class="line1 white"
              />
              <div
                class="line2 black"
              />
              <div
                class="line2 white"
              />
              <div
                class="line3 black"
              />
              <div
                class="line3 white"
              />
            </div>
          </div>
        </div>
        <div
          class="menu-filter"
          id="filter"
        >
          <form
            class="filtering"
            id="ratingSelect"
          >
            <div
              class="movie-rating"
            >
              Rating
              <label>
                <input
                  checked=""
                  id="R"
                  name="stars"
                  type="checkbox"
                  value="R"
                />
                R
              </label>
              <label>
                <input
                  checked=""
                  name="stars"
                  type="checkbox"
                  value="PG-13"
                />
                PG-13
              </label>
              <label>
                <input
                  checked=""
                  name="stars"
                  type="checkbox"
                  value="PG"
                />
                PG
              </label>
              <label>
                <input
                  checked=""
                  name="stars"
                  type="checkbox"
                  value="G"
                />
                G
              </label>
            </div>
            <div
              class="movie-review"
            >
              <label>
                Review Rating
              </label>
              <div
                class="star-buttons"
              >
                <label>
                  <input
                    checked=""
                    name="stars"
                    type="checkbox"
                    value="1"
                  />
                  <span
                    class="icon-full"
                  >
                    ★
                  </span>
                  <span
                    class="icon"
                  >
                    ☆
                  </span>
                  <span
                    class="icon"
                  >
                    ☆
                  </span>
                  <span
                    class="icon"
                  >
                    ☆
                  </span>
                  <span
                    class="icon"
                  >
                    ☆
                  </span>
                </label>
                <label>
                  <input
                    checked=""
                    name="stars"
                    type="checkbox"
                    value="2"
                  />
                  <span
                    class="icon-full"
                  >
                    ★
                  </span>
                  <span
                    class="icon-full"
                  >
                    ★
                  </span>
                  <span
                    class="icon"
                  >
                    ☆
                  </span>
                  <span
                    class="icon"
                  >
                    ☆
                  </span>
                  <span
                    class="icon"
                  >
                    ☆
                  </span>
                </label>
                <label>
                  <input
                    checked=""
                    name="stars"
                    type="checkbox"
                    value="3"
                  />
                  <span
                    class="icon-full"
                  >
                    ★
                  </span>
                  <span
                    class="icon-full"
                  >
                    ★
                  </span>
                  <span
                    class="icon-full"
                  >
                    ★
                  </span>
                  <span
                    class="icon"
                  >
                    ☆
                  </span>
                  <span
                    class="icon"
                  >
                    ☆
                  </span>
                </label>
                <label>
                  <input
                    checked=""
                    name="stars"
                    type="checkbox"
                    value="4"
                  />
                  <span
                    class="icon-full"
                  >
                    ★
                  </span>
                  <span
                    class="icon-full"
                  >
                    ★
                  </span>
                  <span
                    class="icon-full"
                  >
                    ★
                  </span>
                  <span
                    class="icon-full"
                  >
                    ★
                  </span>
                  <span
                    class="icon"
                  >
                    ☆
                  </span>
                </label>
                <label>
                  <input
                    checked=""
                    name="stars"
                    type="checkbox"
                    value="5"
                  />
                  <span
                    class="icon-full"
                  >
                    ★
                  </span>
                  <span
                    class="icon-full"
                  >
                    ★
                  </span>
                  <span
                    class="icon-full"
                  >
                    ★
                  </span>
                  <span
                    class="icon-full"
                  >
                    ★
                  </span>
                  <span
                    class="icon-full"
                  >
                    ★
                  </span>
                </label>
              </div>
            </div>
            <div
              class="sorting"
            >
              <label>
                Sort By :
              </label>
              <label>
                <input
                  name="filter"
                  type="radio"
                  value="recent"
                />
                Most Recent
              </label>
              <label>
                <input
                  name="filter"
                  type="radio"
                  value="old"
                />
                Oldest
              </label>
              <label>
                <input
                  name="filter"
                  type="radio"
                  value="az"
                />
                A-Z
              </label>
              <label>
                <input
                  name="filter"
                  type="radio"
                  value="za"
                />
                Z-A
              </label>
            </div>
            <button
              class="results-btn"
            >
              See Results
            </button>
          </form>
        </div>
      </div>
    </div>
  </body>,
  "container": <div>
    <div>
      <div
        id="hamburger-menu"
      >
        <div
          class="linediv"
        >
          Filter
          <div
            class="linecon"
          >
            <div
              class="line1 black"
            />
            <div
              class="line1 white"
            />
            <div
              class="line2 black"
            />
            <div
              class="line2 white"
            />
            <div
              class="line3 black"
            />
            <div
              class="line3 white"
            />
          </div>
        </div>
      </div>
      <div
        class="menu-filter"
        id="filter"
      >
        <form
          class="filtering"
          id="ratingSelect"
        >
          <div
            class="movie-rating"
          >
            Rating
            <label>
              <input
                checked=""
                id="R"
                name="stars"
                type="checkbox"
                value="R"
              />
              R
            </label>
            <label>
              <input
                checked=""
                name="stars"
                type="checkbox"
                value="PG-13"
              />
              PG-13
            </label>
            <label>
              <input
                checked=""
                name="stars"
                type="checkbox"
                value="PG"
              />
              PG
            </label>
            <label>
              <input
                checked=""
                name="stars"
                type="checkbox"
                value="G"
              />
              G
            </label>
          </div>
          <div
            class="movie-review"
          >
            <label>
              Review Rating
            </label>
            <div
              class="star-buttons"
            >
              <label>
                <input
                  checked=""
                  name="stars"
                  type="checkbox"
                  value="1"
                />
                <span
                  class="icon-full"
                >
                  ★
                </span>
                <span
                  class="icon"
                >
                  ☆
                </span>
                <span
                  class="icon"
                >
                  ☆
                </span>
                <span
                  class="icon"
                >
                  ☆
                </span>
                <span
                  class="icon"
                >
                  ☆
                </span>
              </label>
              <label>
                <input
                  checked=""
                  name="stars"
                  type="checkbox"
                  value="2"
                />
                <span
                  class="icon-full"
                >
                  ★
                </span>
                <span
                  class="icon-full"
                >
                  ★
                </span>
                <span
                  class="icon"
                >
                  ☆
                </span>
                <span
                  class="icon"
                >
                  ☆
                </span>
                <span
                  class="icon"
                >
                  ☆
                </span>
              </label>
              <label>
                <input
                  checked=""
                  name="stars"
                  type="checkbox"
                  value="3"
                />
                <span
                  class="icon-full"
                >
                  ★
                </span>
                <span
                  class="icon-full"
                >
                  ★
                </span>
                <span
                  class="icon-full"
                >
                  ★
                </span>
                <span
                  class="icon"
                >
                  ☆
                </span>
                <span
                  class="icon"
                >
                  ☆
                </span>
              </label>
              <label>
                <input
                  checked=""
                  name="stars"
                  type="checkbox"
                  value="4"
                />
                <span
                  class="icon-full"
                >
                  ★
                </span>
                <span
                  class="icon-full"
                >
                  ★
                </span>
                <span
                  class="icon-full"
                >
                  ★
                </span>
                <span
                  class="icon-full"
                >
                  ★
                </span>
                <span
                  class="icon"
                >
                  ☆
                </span>
              </label>
              <label>
                <input
                  checked=""
                  name="stars"
                  type="checkbox"
                  value="5"
                />
                <span
                  class="icon-full"
                >
                  ★
                </span>
                <span
                  class="icon-full"
                >
                  ★
                </span>
                <span
                  class="icon-full"
                >
                  ★
                </span>
                <span
                  class="icon-full"
                >
                  ★
                </span>
                <span
                  class="icon-full"
                >
                  ★
                </span>
              </label>
            </div>
          </div>
          <div
            class="sorting"
          >
            <label>
              Sort By :
            </label>
            <label>
              <input
                name="filter"
                type="radio"
                value="recent"
              />
              Most Recent
            </label>
            <label>
              <input
                name="filter"
                type="radio"
                value="old"
              />
              Oldest
            </label>
            <label>
              <input
                name="filter"
                type="radio"
                value="az"
              />
              A-Z
            </label>
            <label>
              <input
                name="filter"
                type="radio"
                value="za"
              />
              Z-A
            </label>
          </div>
          <button
            class="results-btn"
          >
            See Results
          </button>
        </form>
      </div>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
